### PR TITLE
pmemcheck: fix flush registering bug, change logs to hex format

### PR DESCRIPTION
**Pull request by tomaszkapela**
Fixes the flush problem, changes log output format and reorders some tests. The logging of flushes of non-dirty stores is controlled by a CLO flag.

### CODE REVIEW

File: pmemcheck/pmc_main.c [hunk 1]
```diff
@@ -111,6 +111,8 @@ static struct pmem_ops {
     /** Toggles summary printing. */
     Bool print_summary;
 
+    Bool check_flush;
```
Comment (krzycz) [inline pmemcheck/pmc_main.c:115]: Comment?

File: pmemcheck/pmc_main.c [hunk 2]
```diff
@@ -736,13 +738,20 @@ do_flush(UWord base, UWord size)
...
-           /* flushing non dirty store - probably an issue, record */
-           pmem.flush_errors[pmem.flush_errors_reg] = VG_(malloc)("pmc.main"
-                   ".cpci.2", sizeof (struct pmem_st));
-           *(pmem.flush_errors[pmem.flush_errors_reg]) = *being_flushed;
+       /* not an interesting entry, flush doesn't matter */
+       if (cmp_pmem_st(&flush_info, being_flushed) != 0) {
+           continue;
+       }
+
+        /* flushing a non-dirty store */
```
Comment (krzycz) [inline pmemcheck/pmc_main.c]: Fix indentation, please.

File: pmemcheck/tests/common.h [hunk 3]
```diff
@@ -12,20 +12,14 @@
...
-#include <signal.h>
 #include <fcntl.h>
 #include <unistd.h>
 #include <errno.h>
-#include <stdint.h>
 #include <sys/mman.h>
-#include <sys/stat.h>
 
-#include "pmemcheck/pmemcheck.h"
+#include "../pmemcheck.h"
```
Comment (krzycz) [inline pmemcheck/tests/common.h:22]: Don't like this style of including. What about adding `-I..` to the compiler options.

File: pmemcheck/tests/common.h [hunk 4]
```diff
@@ -63,6 +57,7 @@ make_map_tmpfile(size_t size)
 
     close(fd);
 
+    /* does not log the file */
```
Comment (krzycz) [inline pmemcheck/tests/common.h]: Not sure what it means.

File: pmemcheck/tests/flush_check.stderr.exp [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Number of stores not made persistent: 1
+Stores not made persistent properly:
+[0]    at 0x........: main (flush_check.c:27)
+	Address: 0x........	size: 8	state: COMMITED
+Total memory not made persistent: 8
+
+Number of flushes on non-dirty stores: 3
+Stores flushed when not dirty:
```
Comment (krzycz) [inline pmemcheck/tests/flush_check.stderr.exp]: Actually, these are not "non-dirty stores". There were no stores, but only flushes. So, maybe we should rather report it as "Cache lines flushed when non-dirty".

File: pmemcheck/tests/logging_related/logging.c [hunk 6]
```diff
@@ -12,7 +12,7 @@
  * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
  * more details.
  */
-#include "common.h"
+#include "../common.h"
```
Comment (krzycz) [inline pmemcheck/tests/logging_related/logging.c:15]: `-I..` ?

File: pmemcheck/tests/common.h (hunk 3 above)
Comment (tomaszkapela) [inline pmemcheck/tests/common.h:22]: This is the standard way of including headers in valgrind tests.

**Pull request merged**